pkg/tc/filterset_test.go

```diff
@@ -0,0 +1,511 @@
+package tc_test
+
+import (
+	"github.com/Mellanox/multi-networkpolicy-tc/pkg/tc"
+	tctypes "github.com/Mellanox/multi-networkpolicy-tc/pkg/tc/types"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("FilterSetImpl tests", func() {
+	var filterSet tc.FilterSet
+
+	BeforeEach(func() {
+		filterSet = tc.NewFilterSetImpl()
+	})
+
+	Context("FilterSet.Add()", func() {
+		It("Adds filter to FilterSet", func() {
+			filters := []tctypes.Filter{
+				tctypes.NewFlowerFilterBuilder().
+					WithProtocol(tctypes.FilterProtocolIP).
+					WithAction(
+						tctypes.NewGenericActionBuiler().
+							WithDrop().
+							Build()).
+					Build(),
+				tctypes.NewFlowerFilterBuilder().
+					WithProtocol(tctypes.FilterProtocolIP).
+					WithAction(
+						tctypes.NewGenericActionBuiler().
+							WithPass().
+							Build()).
+					Build(),
+			}
+			for i := range filters {
+				filterSet.Add(filters[i])
+			}
+
+			filterList := filterSet.List()
+			Expect(filterList).To(HaveLen(2))
+			for i := range filterList {
+				Expect(filterList[i].Equals(filters[i])).To(BeTrue())
```

Comment:
nit: the fact that filterSet presevers order is a side effect of the implementation, not the part of the contract. IMO, it is better to check with `Expect(filterList).To(ContainElements(filters))`